### PR TITLE
Add pv argument to get_all_pvs that allows a glob pattern.

### PR DESCRIPTION
**Pull request by willrogers**

### CODE REVIEW

**Comment by aawdls** [comment]
It looks sensible enough. I will test it, and assuming it works, rebase and merge.

**Comment by aawdls** [comment]
OK I think I know the reason why you postponed this. I get the same test failure that you do.

Adding the extra argument here makes this the only use with more than one URL parameter. Because of the non-deterministic ordering in dictionaries, two successive calls to `AaRestClient._construct_url` output the parameters in different orders. This causes the test to fail.

My preferred way to fix this would be to change `_construct_url` to accept an OrderedDict for the URL parameters, instead of using `**kwargs`.

I guess at the time you questioned whether this was worth it. I think it probably is so I'll go ahead and do it for neatness' sake.

**Comment by willrogers** [comment]
That sounds right. Deterministic URLs make sense I think.

File: aa/utils.py [hunk 1]
```diff
@@ -121,3 +123,15 @@ def set_up_logging(fmt=LOG_FORMAT, level=LOG_LEVEL):
 
     """
     logging.basicConfig(format=fmt, level=level, datefmt="%Y-%m-%d %I:%M:%S")
+
+
+def dict_to_tuples(input_dict: dict) -> List[tuple]:
```
Comment (willrogers) [inline aa/utils.py:128]: `Dict[str, str]` and `List[Tuple[str, str]]`?

**Comment by willrogers** [comment]
Thanks, looks good.

**Pull request merged**